android-components/components/feature/media/src/main/java/mozilla/components/feature/media/fullscreen/MediaSessionFullscreenFeature.kt

```diff
@@ -35,51 +35,53 @@ class MediaSessionFullscreenFeature(
             flow.map {
                 it.tabs + it.customTabs
             }.map { tab ->
-                tab.filter { it.mediaSessionState != null && it.mediaSessionState!!.fullscreen }
-            }.distinctUntilChanged().collect { states ->
-                processFullscreen(states)
-                processDeviceSleepMode(states)
+                tab.firstOrNull { it.mediaSessionState != null && it.mediaSessionState!!.fullscreen }
```

Comment:
I'm not sure this is worth it, but we could technically get rid of the force unwrap (`!!`) with something like:

```
                    with (it.mediaSessionState) {
                        (this != null) && this.fullscreen
                    }   
```

or
```
                tab.firstOrNull {
                    val mediaState = it.mediaSessionState
                    mediaState != null && mediaState.fullscreen
                }
```

Up to you 🙂

Reply:
Actually I think this could be even better? 
```suggestion
                tab.firstOrNull { it.mediaSessionState?.fullscreen == true }
```

Reply:
Yes, that's perfect!